$NetBSD: patch-ce,v 1.3 2002/10/15 10:47:00 agc Exp $

Robert Elz's PR 18664 contains the rationale for this patch:

        A bug report (from Scott Lipcon) was submitted elsewhere:
http://savannah.nongnu.org/bugs/?func=detailbug&bug_id=1393&group_id=2166
        and then today copied to the nmh-workers list.

        Basically, using "sortm -textfield subject" where a message
        in the folder contains "Subject:N" where "N" is alphanumeric
        will cause sortm to dump core (the "korean spam" aspect of the
        mail that caused the original bug report is irrelevant).

>How-To-Repeat:
        Use nmh.   Put a message in a folder that contains "Subject:1234"
        and then run
                sortm +whatever-folder -textfield subject
        and watch for the core dump.

        Or, examine the code and be amazed ...

--- uip/sortm.c	2000/02/04 20:28:24	1.3
+++ uip/sortm.c	2002/10/15 10:38:32
@@ -439,9 +439,12 @@
 			if(uprf(cp, "re:"))
 			    cp += 2;
 			else {
+#if 0
 			    if (isalnum(c))
 				*cp2++ = isupper(c) ? tolower(c) : c;
+#else
 			    break;
+#endif
 			}
 		    }
 		    cp++;
